src/smc-webapp/editor.coffee

```diff
@@ -933,7 +932,7 @@ class CodeMirrorEditor extends FileEditor
         if opts.read_only
             @set_readonly_ui()
 
-        if @filename.slice(@filename.length-7) == '.sagews'
+        if misc.filename_extension(@filename)?.toLowerCase() == 'sagews'
```

Comment:
I have to wonder if filename_extension and filename_extension_notilde should just always do `toLowerCase()` ?